crates/aptos/src/op/key.rs

```diff
@@ -208,25 +221,34 @@ impl CliCommand<HashMap<&'static str, PathBuf>> for GenerateKey {
             },
             KeyType::Ed25519 => {
                 // If no vanity prefix specified, generate a standard Ed25519 private key.
-                let private_key = if self.vanity_prefix.vanity_prefix.is_none() {
+                let private_key = if self.vanity_prefix.is_none() {
                     keygen.generate_ed25519_private_key()
                 } else {
                     // If a vanity prefix is specified, generate vanity Ed25519 account from it.
                     generate_vanity_account_ed25519(
-                        self.vanity_prefix.vanity_prefix.clone().unwrap().as_str(),
+                        self.vanity_prefix.clone().unwrap().as_str(),
+                        self.vanity_multisig,
                     )?
                 };
-                // Store CLI result map from key save operation, to append vanity address if needed.
+                // Store CLI result from key save operation, to append vanity address(es) if needed.
                 let mut result_map = self.save_params.save_key(&private_key, "ed25519").unwrap();
-                // If a vanity prefix was specified:
-                if self.vanity_prefix.vanity_prefix.is_some() {
-                    // Get account address.
+                if self.vanity_prefix.is_some() {
                     let account_address = account_address_from_public_key(
                         &ed25519::Ed25519PublicKey::from(&private_key),
-                    )
-                    .to_hex_literal();
-                    // Put account address in PathBuf so it can be displayed from typed CLI result.
-                    result_map.insert("Account Address:", PathBuf::from(account_address));
+                    );
+                    // Store account address in a PathBuf so it can be displayed in CLI result.
+                    result_map.insert(
+                        "Account Address:",
+                        PathBuf::from(account_address.to_hex_literal()),
+                    );
+                    if self.vanity_multisig {
```

Comment:
why are we adding this if multisig is true?

Reply:
Yeah the flow for multisig here is quite strange, as it technically doesn't create the multisig account itself. The user would then need to create one later with the right seed (0).